wolfcrypt/src/sha512.c

```diff
@@ -1462,6 +1503,10 @@ int wc_Sha384Update(Sha384* sha384, const byte* data, word32 len)
 
 int wc_Sha384Final(Sha384* sha384, byte* hash)
 {
+    if (sha384 == NULL || hash == NULL) {
+        return BAD_FUNC_ARG;
+    }
+
     byte* local = (byte*)sha384->buffer;
```

Comment:
move variable declarations to beginning of statement or change to "declaration then assignement"